cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueClientFactory.java

```diff
@@ -42,35 +44,37 @@
   private final HBaseQueueAdmin queueAdmin;
   private final HBaseStreamAdmin streamAdmin;
   private final HBaseQueueUtil queueUtil;
+  private final HBaseTableUtil hBaseTableUtil;
 
   @Inject
-  public HBaseQueueClientFactory(CConfiguration cConf, Configuration hConf,
+  public HBaseQueueClientFactory(CConfiguration cConf, Configuration hConf, HBaseTableUtil hBaseTableUtil,
                                  QueueAdmin queueAdmin, HBaseStreamAdmin streamAdmin) {
     this.cConf = cConf;
     this.hConf = hConf;
     this.queueAdmin = (HBaseQueueAdmin) queueAdmin;
     this.streamAdmin = streamAdmin;
     this.queueUtil = new HBaseQueueUtilFactory().get();
+    this.hBaseTableUtil = hBaseTableUtil;
   }
 
   // for testing only
-  String getTableName(QueueName queueName) {
-    return (queueName.isStream() ? streamAdmin : queueAdmin).getActualTableName(queueName);
+  TableId getTableId(QueueName queueName) {
+    return (queueName.isStream() ? streamAdmin : queueAdmin).getDataTableId(queueName);
   }
 
   // for testing only
-  String getConfigTableName(QueueName queueName) {
-    return (queueName.isStream() ? streamAdmin : queueAdmin).getConfigTableName(queueName);
+  TableId getConfigTableId(QueueName queueName) {
+    return HBaseQueueAdmin.getConfigTableId(queueName);
   }
 
   @Override
   public QueueConsumer createConsumer(QueueName queueName,
                                       ConsumerConfig consumerConfig, int numGroups) throws IOException {
     HBaseQueueAdmin admin = ensureTableExists(queueName);
-    HTable configTable = createHTable(admin.getConfigTableName(queueName));
+    HTable configTable = createHTable(HBaseQueueAdmin.getConfigTableId(queueName));
```

Comment:
actually, what do you feel about having this be a non-static method in HBaseQueueAdmin. All it needs is the namespace, which is available in the ``queueName``.

Although it may mean that ``HBaseQueueAdmin`` - the admin for a data table knows about the config table as well? Not sure.